packages/adapter-drizzle/src/planetscale/index.ts

```diff
@@ -0,0 +1,248 @@
+  host: process.env["DATABASE_HOST"],
+  username: process.env["DATABASE_USERNAME"],
+  password: process.env["DATABASE_PASSWORD"],
+})
+
+export const db = drizzle(connection)
+export type DbClient = typeof db
+
+export const defaultSchema = { users, accounts, sessions, verificationTokens }
+export type DefaultSchema = typeof defaultSchema
+interface CustomSchema extends DefaultSchema { }
+
+export function PlanetScaleAdapter(
+  client: DbClient,
+  schema?: Partial<CustomSchema>
+): Adapter {
+  const { users, accounts, sessions, verificationTokens } = {
+    users: schema?.users ?? defaultSchema.users,
+    accounts: schema?.accounts ?? defaultSchema.accounts,
+    sessions: schema?.sessions ?? defaultSchema.sessions,
+    verificationTokens:
+      schema?.verificationTokens ?? defaultSchema.verificationTokens,
+  }
+
+  return {
+    createUser: async (data) => {
+      const id = crypto.randomUUID()
+
+      await client.insert(users).values({ ...data, id })
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, id))
+        .then((res) => res[0])
+    },
+    getUser: async (data) => {
+      return (
+        client
+          .select()
+          .from(users)
+          .where(eq(users.id, data))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    getUserByEmail: async (data) => {
+      return (
+        client
+          .select()
+          .from(users)
+          .where(eq(users.email, data))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    createSession: async (data) => {
+      await client.insert(sessions).values(data)
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then((res) => res[0])
+    },
+    getSessionAndUser: async (data) => {
+      return (
+        client
+          .select({
+            session: sessions,
+            user: users,
+          })
+          .from(sessions)
+          .where(eq(sessions.sessionToken, data))
+          .innerJoin(users, eq(users.id, sessions.userId))
+          .then((res) => res[0]) ?? null
+      )
+    },
+    updateUser: async (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      await client.update(users).set(data).where(eq(users.id, data.id))
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, data.id))
+        .then((res) => res[0])
+    },
+    updateSession: async (data) => {
+      await client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+
+      return client
+        .select()
+        .from(sessions)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .then((res) => res[0])
+    },
+    linkAccount: async (rawAccount) => {
+      await client
+        .insert(accounts)
+        .values(rawAccount)
+        .then((res) => res.rows[0])
```

Comment:
This is failing for me trying Google Auth. The `rawAccount.id_token` is bigger than 255 chars. Is 1185 chars. So maybe this column `id_token` should be a text column.

Reply:
Will be dropping PlanetScale to cut scope so we can ship. We should be able to add PlanetScale back in the future!